pkg/sidecar/k8s_instance.go

```diff
@@ -82,6 +83,10 @@ func (d *K8sInstanceManager) Close() error {
 }
 
 func (d *K8sInstanceManager) manageContainer(ctx context.Context, container *dockermanager.Container) (inst *Instance, err error) {
+	// TODO: sidecar is racing to modify container network with CNI and pod getting ready
+	// we should probably adjust this function to be called when a pod is in `1/1 Ready` state, and not just listen on the docker socket
```

Comment:
Will the test end up starting before then? I assume not. If so, we could wait for the test to configure the network.

Alternatively, I wonder if we should be waiting for a different docker event.

Reply:
@Stebalien I think we should be waiting for the kubernetes API to report that the pod is `Ready` or (`Running 1/1`), but this is yet to be tested.

Reply:
We can do that, I think. I'm slightly concerned at the coordination cost but it shouldn't be terrible.